rs/canister_sandbox/src/replica_controller/sandboxed_execution_controller.rs

```diff
@@ -2299,4 +2304,112 @@ mod tests {
         );
         assert_eq!(0, partitioned_backends.2.len());
     }
+
+    #[test]
+    fn monitor_and_evict_thread_is_spawned() {
+        let active = 1;
+        let spawn_monitor_thread = true;
+        let (controller, _dir, _path) =
+            sandboxed_execution_controller_dir_and_path(active, spawn_monitor_thread);
+        assert!(controller.stop_monitoring_thread.send(true).is_ok());
+
+        let spawn_monitor_thread = false;
+        let (controller, _dir, _path) =
+            sandboxed_execution_controller_dir_and_path(active, spawn_monitor_thread);
+        assert!(controller.stop_monitoring_thread.send(true).is_err());
+    }
+
+    #[test]
+    #[cfg(not(all(target_arch = "aarch64", target_vendor = "apple")))]
+    fn monitor_and_evict_thread_collects_rss() {
+        let active = 1;
+        let spawn_monitor_thread = false;
+        let (mut controller, _dir, _path) =
+            sandboxed_execution_controller_dir_and_path(active, spawn_monitor_thread);
+        add_controller_backends(&mut controller, 0, active, 0, 0);
+        let stats = get_sandbox_process_stats(&controller.backends);
+        assert_eq!(stats.len(), active);
+        assert_ne!(stats[0].1.pid, 0);
+        assert!(stats[0].2.last_used <= Instant::now());
+        assert!(stats[0].2.last_used >= Instant::now() - Duration::from_secs(1_000));
+        assert_eq!(stats[0].2.rss, DEFAULT_SANDBOX_PROCESS_RSS);
+
+        let spawn_monitor_thread = true;
+        let (mut controller, _dir, _path) =
+            sandboxed_execution_controller_dir_and_path(active, spawn_monitor_thread);
+        add_controller_backends(&mut controller, 0, active, 0, 0);
+
+        // Trigger the monitoring and wait for the monitoring results.
+        let _ = controller.stop_monitoring_thread.send(false);
```

Comment:
Maybe we should document the `stop_monitoring_thread` field to note that sending `false` will trigger the monitoring loop? Just because it's not so obvious.